File: src/Phoenix.Models/Plcs/Rvds/Dto/Rvd145Dto.cs
using Phoenix.Models.Base.Dto;

namespace Phoenix.Models.Plcs.Rvds.Dto
{
   public sealed class Rvd145Dto : PlcDtoBase
   {
      public float OutsideTemp { get; init; }
      public float ChHighInletPresure { get; init; }
      public short Alarm { get; init; }

      public float Ch1HighOutletTemp { get; init; }
      public float Ch1LowInletTemp { get; init; }
      public float Ch1LowOutletPresure { get; init; }
      public float Ch1HeatCurveTemp { get; init; }
      public bool Ch1PumpStatus { get; init; }
      public bool Ch1Status { get; init; }

      public float DhwTemp { get; init; }
      public float DhwTempSet { get; init; }
      public float DhwCirculationTemp { get; init; }
      public bool DhwPumpStatus { get; init; }
      public bool DhwStatus { get; init; }
   }
}

File: src/Phoenix.Web/src/models/api/plcs/rvds/dto/rvd145Dto.ts
import { PlcDtoBase } from "../../../base/dto/plcDtoBase";

export interface Rvd145Dto extends PlcDtoBase {
   outsideTemp: number;
   chHighInletPresure: number;
   alarm: number;
   ch1HighOutletTemp: number;
   ch1LowInletTemp: number;
   ch1LowOutletPresure: number;
   ch1HeatCurveTemp: number;
   ch1PumpStatus: boolean;
   ch1Status: boolean;
   dhwTemp: number;
   dhwTempSet: number;
   dhwCirculationTemp: number;
   dhwPumpStatus: boolean;
   dhwStatus: boolean;
}
